tests/init_spec.lua

```diff
@@ -47,13 +51,22 @@ describe("is_test_file", function()
 
   async.it("does not match test in repo with jest", function()
     vim.api.nvim_set_current_dir("./spec-jest")
-    assert.is.falsy(plugin.is_test_file("./spec-jest/basic.test.ts"))
+    assert.is.falsy(plugin.is_test_file("./basic.test.ts"))
   end)
 
   async.it("matches vitest files in monorepo", function()
     vim.api.nvim_set_current_dir("./spec-monorepo")
-    assert.is.truthy(plugin.is_test_file("./spec-monorepo/packages/example/basic.test.ts"))
-    assert.is.truthy(plugin.is_test_file("./spec-monorepo/apps/todo/todo.test.tsx"))
+    assert.is.truthy(plugin.is_test_file("./packages/example/basic.test.ts"))
```

Comment:
Note that the relative path here was wrong due to the `nvim_set_current_dir`. It was causing `match_root_pattern` to look for `wholepath/spec-jest/spec-jest/basic.test.ts`, and by luck it resolved upwards until it hit `spec-monorepo` which has vitest, but would miss `wholepath/spec-monorepo/packages/example/package.json` which might have had vitest.